File: docs/extract_docx.js
const fs = require('fs');
const AdmZip = require('adm-zip');
const xml2js = require('xml2js');

async function extractDocx(docxPath) {
    try {
        // Read the DOCX file as a ZIP
        const zip = new AdmZip(docxPath);
        
        // Get the document.xml entry
        const docEntry = zip.getEntry('word/document.xml');
        if (!docEntry) {
            console.error('Error: word/document.xml not found in DOCX');
            return;
        }
        
        const xmlContent = zip.readAsText(docEntry);
        
        // Parse XML
        const parser = new xml2js.Parser();
        const result = await parser.parseStringPromise(xmlContent);
        
        // Extract document root
        const document = result['w:document'];
        if (!document || !document['w:body']) {
            console.error('Error: Invalid document structure');
            return;
        }
        
        const body = document['w:body'][0];
        
        // Process all elements (paragraphs and tables)
        if (body['w:p']) {
            body['w:p'].forEach(para => {
                let paraText = '';
                if (para['w:r']) {
                    para['w:r'].forEach(run => {
                        if (run['w:t'] && run['w:t'][0]) {
                            paraText += run['w:t'][0];
                        }
                    });
                }
                if (paraText.trim()) {
                    console.log(paraText);
                }
            });
        }
        
        if (body['w:tbl']) {
            body['w:tbl'].forEach(table => {
                console.log('\n--- TABLE ---');
                if (table['w:tr']) {
                    table['w:tr'].forEach(row => {
                        const cells = [];
                        if (row['w:tc']) {
                            row['w:tc'].forEach(cell => {
                                let cellText = '';
                                if (cell['w:p']) {
                                    cell['w:p'].forEach(para => {
                                        if (para['w:r']) {
                                            para['w:r'].forEach(run => {
                                                if (run['w:t'] && run['w:t'][0]) {
                                                    cellText += run['w:t'][0];
                                                }
                                            });
                                        }
                                    });
                                }
                                cells.push(cellText);
                            });
                        }
                        console.log(cells.join(' | '));
                    });
                }
                console.log('--- END TABLE ---\n');
            });
        }
    } catch (error) {
        console.error('Error:', error.message);
    }
}

const docxFile = process.argv[2] || 'c:\\Users\\mnaco\\Github\\tontApp\\docs\\TontinesApp_UserStories_MVP.docx';
extractDocx(docxFile);

File: docs/extract_docx.py
#!/usr/bin/env python3
"""Extract ALL text content from a DOCX file"""

import sys
from pathlib import Path

# Try to import python-docx, install if needed
try:
    from docx import Document
except ImportError:
    import subprocess
    print("Installing python-docx...", file=sys.stderr)
    subprocess.check_call([sys.executable, "-m", "pip", "install", "python-docx", "-q"])
    from docx import Document

def extract_docx_content(docx_path):
    """Extract all paragraphs and table content from DOCX file"""
    
    doc = Document(docx_path)
    
    # Extract all paragraphs
    for para in doc.paragraphs:
        if para.text.strip():  # Only print non-empty paragraphs
            print(para.text)
    
    # Extract all tables
    for table in doc.tables:
        print("\n--- TABLE ---")
        for row in table.rows:
            cells_text = []
            for cell in row.cells:
                # Extract text from cell (may contain multiple paragraphs)
                cell_text = " ".join([p.text for p in cell.paragraphs if p.text.strip()])
                cells_text.append(cell_text)
            print(" | ".join(cells_text))
        print("--- END TABLE ---\n")

if __name__ == "__main__":
    docx_file = r"c:\Users\mnaco\Github\tontApp\docs\TontinesApp_UserStories_MVP.docx"
    
    if not Path(docx_file).exists():
        print(f"Error: File not found: {docx_file}", file=sys.stderr)
        sys.exit(1)
    
    extract_docx_content(docx_file)
